kernel/malloc.c

```diff
@@ -70,7 +70,7 @@ void *solo5_calloc(size_t, size_t) __attribute__ ((alias ("calloc")));
 void *solo5_realloc(void *, size_t) __attribute__ ((alias ("realloc")));
 
 /* disable null-pointer-arithmetic warning on clang */
-#if defined(__clang__)
+#if defined(__clang__) && !defined(__OpenBSD__) 
 #pragma clang diagnostic ignored "-Wunknown-pragmas"
```

Comment:
for older (3.8, 4.0) clang on FreeBSD, ignoring the `unknown-pragmas` -- as done here, was sufficient for me.  doesn't this work on OpenBSD?

Reply:
@hannesm before you commented i added the change
 `#if defined(__clang__)  &&  __clang_major > 5`
which seems better as it what the old code had for earlier version of clang
`#if defined(__clang__) && !defined(__OpenBSD__) `

Reply:
yes, but it needs to be `__clang_major__` (i.e. you missed the `__` at the end) AFAICT -- would be great if you can try #241 instead of the `!defined(__OpenBSD__)` :)

Reply:
yes #241 works perfectly. and merged into my branch.